### DISCUSSION TITLE
Kotlin enums do NOT support reassignment!

### DISCUSSION THREAD

**Discussion by davidakoontz** [skiptools/skip#264] [Q&A] [answered]
I just ran into this problem.  Wonder if anyone has worked around it could point the way?

In Swift I've a:  Enum Step with a next()...

```
enum Step: CaseIterable {
    case Step_0, Step_1, Step_2, Step_3, Step_4, Step_5
    mutating func next() {
        switch self {
            case .Step_0:
                self = .Step_1
        // etc.
        
```

This creates an Error in Skip:  Kotlin enums do NOT support reassignment!

What's a good way to work out of this conundrum ?

**Comment by aabewhite** [comment, marked as answer]
Yes, unfortunately enums do not support mutating functions. We recommend moving to a next() function that returns the next value and updating call sites to use `step = step.next()`

**Reply by davidakoontz** [reply]
AHHH... I think I get it... return the next step - but DO NOT assign it in the next() function!  Yeah that might work just perfect!

    // Kotlin enums do not support reassignment
    // so try just returning the next...
    func next() -> Step {
        switch self {
            case .Step_0:
                return .Step_1
            case .Step_1:
                return .Step_2
            case .Step_2:
                return .Step_3
            case .Step_3:
                return .Step_4
            case .Step_4:
                return .Step_5
            case .Step_5:
                return .Step_0      // recycles to beginning
        }
    }
    

        .onDisappear {
            appState.onStep = appState.onStep.next()
            print("Exit Try PMap View - Step is \(appState.onStep)")
        }

**Reply by davidakoontz** [reply]
"but what about class boundries?" 
"is the encapsulated Enum going to leak all over the place"

tell all those professors of CS whispering in my ears... go away - Ive got real coding to do...
